### DISCUSSION TITLE
Can't get pickFiles method on ui.upload to work in Safari (works in Chrome)

### DISCUSSION THREAD

**Discussion by supercurses** [zauberzeug/nicegui#3885] [Q&A]
### Question

I must be missing something obvious.

```py
upload = ui.upload(auto_upload=True, on_upload=lambda e: ui.notify(f'{len(e.content.read())} bytes')).classes('hidden')
ui.button('Upload', on_click=lambda: upload.run_method('pickFiles'))
```

Does nothing, no errors, no file dialog is spawned. (context, I just want to create a less UI heavy file upload component)

**Comment by supercurses** [comment]
Well, there we go, it's been a while since I've said this: works in Chrome, doesn't work in Safari.

**Comment by falkoschindler** [comment]
Hi @supercurses,

It looks like Safari hesitates to show a file picker dialog if the action isn't triggered by the user, but - in the browser's point of view - via SocketIO from a server.

We can avoid this problem by calling `pickFiles()` directly on the client:
```py
ui.button('Upload').on('click', js_handler=f'() => getElement({upload.id}).$refs.qRef.pickFiles()')
```

Don't ask me about the part `.$refs.qRef.`... It's always a bit of trial and error and asking your preferred AI to get this straight. But at least it works.
